src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -282,7 +287,7 @@ private static boolean isSchematicSizeValid(@NotNull final String structureName)
         final byte[] data = Structure.getStreamAsByteArray(Structure.getStream(structureName));
         final byte[] compressed = Structure.compress(data);
 
-        if (compressed != null && compressed.length > maxSize)
+        if (compressed == null)
         {
             Log.getLogger().warn("Structure " + structureName + " is " + compressed.length + " bytes when compress, maximum allowed is " + maxSize + " bytes.");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "compressed" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)